lib/ansible/modules/cloud/amazon/code_build.py

```diff
@@ -0,0 +1,338 @@
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to update project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def delete_project(client, name, module):
+    try:
+        resp = client.delete_project(name=name)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable delete project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to delete project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def describe_project(client, name, module):
+    project = {}
+    try:
+        projects = client.batch_get_projects(names=[name])['projects']
+        if len(projects) > 0 and isinstance(projects[0], dict):
+            project = projects[0]
+        return project
+    except botocore.exceptions.ClientError as e:
+        return project
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Error when calling client.batch_get_projects {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(required=True, type='str'),
+        description=dict(required=False, type='str'),
+        source=dict(required=True, type='dict'),
+        artifacts=dict(required=True, type='dict'),
+        cache=dict(required=False, type='dict'),
+        environment=dict(required=True, type='dict'),
+        service_role=dict(required=False, type='str'),
+        timeout_in_minutes=dict(required=False, type='int', default=60),
+        encryption_key=dict(required=False, type='str'),
+        tags=dict(required=False, type='list'),
+        vpc_config=dict(required=False, type='dict'),
+        state=dict(required=False, type='str', choices=['present', 'absent'], default='present')
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required.')
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    client_conn = boto3_conn(module, conn_type='client', resource='codebuild', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+
+    state = module.params.get('state')
+    changed = False
+
+    if state == 'present':
+        project_result = create_or_update_project(
+            client=client_conn,
+            params=module.params,
+            module=module)
+        if project_result['updated']:
+            changed = True
+    elif state == 'absent':
+        project_result = delete_project(client=client_conn, name=module.params['name'], module=module)
+        changed = True
```

Comment:
It looks like changed will be True whether or not the project existed to be deleted.